subscriber/handler.py

```diff
@@ -16,6 +17,38 @@ def run(event, context):
     name = context.function_name
     logger.info("Your httpApi function " + name + " ran at " + str(current_time))
 
+    def respond_with(status_code, message = '', topics = []):
+        headers = event.get("headers")
+        if headers is not None:
+            accept_header = headers.get("accept") or headers.get("Accept")
+            referer_header = headers.get("referer") or headers.get("Referer")
+            if ['application/json', 'text/json'].count(accept_header) > 0:
+                logger.debug(f"Returning JSON response with status code: {status_code}.")
+                return {
+                    "statusCode": status_code,
+                    "headers": {
+                        "Content-Type": "application/json"
+                    },
+                    "body": json.dumps({"message": message, "topics": topics})
+                }
+            elif referer_header is not None:
+                logger.debug(f"Returning redirect response.")
+                return {
+                    "statusCode": 302,
+                    "headers": {
+                        "Location": f"{referer_header}?message={quote_plus(message)}&topics={','.join(topics)}"
```

Comment:
@fayez-nazzal Can the referer header contain query strings, e.g. from the user search? In this case we need to parse it properly.

Reply:
@hammady Yes, the search action adds query strings, but I'm not sure if it will be included in the query params, let me give it a try.

Reply:
@hammady Nope, turned out the referer header will only have the origin.

Reply:
I have implemented qs preserving nevertheless :)